### PR TITLE
CIRC-1432 Actual cost - close aged to lost loan as paid when no processing fee should be charged

### PR DESCRIPTION
**Pull request by viktor-draban**
Resolves : [https://issues.folio.org/browse/CIRC-1432](https://issues.folio.org/browse/CIRC-1432)

## Purpose
The processing of aged to lost actual cost items should be skipped when 'Charge lost item processing fee if item aged to lost by system?' is set to `No` in the Lost item policy (except `lostItemHasBeenBilled` should be changed to `True`).

## Approach
Additional step is added to `ChargeLostFeesWhenAgedToLostService` service that excludes such loans from the list of loans that fees/fines or actual cost records will be created for.

### CODE REVIEW

File: src/test/java/api/handlers/CloseAgedToLostLoanWhenLostItemFeesAreClosedApiTests.java [hunk 1]
```diff
@@ -66,6 +69,25 @@ void shouldCloseLoanWhenAllFeesClosed() {
     assertThat(loanClosedEvents.get(0), isValidLoanClosedEvent(loan.getJson()));
   }
 
+  @Test
+  void shouldProcessingFeeBeingBilledWhenSetActualCost() {
```
Comment (roman-barannyk) [inline src/test/java/api/handlers/CloseAgedToLostLoanWhenLostItemFeesAreClosedApiTests.java:73]:
```suggestion
  void processingFeeShouldBeChargedWhenSetActualCost() {
```

File: src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java [hunk 2]
```diff
@@ -125,10 +125,26 @@ private CompletableFuture<Result<Void>> chargeLostFeesForLoans(
     Result<List<LoanToChargeFees>> loansToChargeFeesResult) {
 
     return loansToChargeFeesResult
+      .next(this::excludeActualCostItemsWithNoLostItemProcessingFee)
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java:128]: don't exclude, keep processing, close loans

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java [hunk 3]
```diff
@@ -125,8 +125,14 @@ private CompletableFuture<Result<Void>> chargeLostFeesForLoans(
...
-      .after(loansToChargeFees -> allOf(loansToChargeFees, this::chargeLostFees))
-      .thenApply(r -> r.map(ignored -> null));
+      .after(loans -> allOf(loans, this::processLoan))
+      .thenApply(Result::mapEmpty);
+  }
+
+  private CompletableFuture<Result<Void>> processLoan(LoanToChargeFees loan) {
+    return loan.shouldCloseLoanWhenActualCostUsed()
+      ? updateBillingInfoAndCloseLoanAsLostAndPaid(loan)
+      : chargeLostFees(loan);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java:135]: Can we flip the condition?
```suggestion
    return loan.shouldChargeLostItemFess()
      ? chargeLostFees(loan)
      : updateBillingInfoAndCloseLoanAsLostAndPaid(loan);
```

File: src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java [hunk 4]
```diff
@@ -323,4 +329,13 @@ private CompletableFuture<Result<Loan>> closeLoanAsLostAndPaid(LoanToChargeFees
 
     return storeLoanAndItem.updateLoanAndItemInStorage(loan);
   }
+
+  private CompletableFuture<Result<Void>> updateBillingInfoAndCloseLoanAsLostAndPaid(
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/circulation/services/agedtolost/ChargeLostFeesWhenAgedToLostService.java:333]: we don't update billing info in this method

**Review by OleksandrVidinieiev — APPROVED** [review]

**Pull request merged**